arrow-json/src/reader.rs

```diff
@@ -526,16 +528,17 @@ fn collect_field_types_from_object(
 /// The reason we diverge here is because we don't have utilities to deal with JSON data once it's
 /// interpreted as Strings. We should match Spark's behavior once we added more JSON parsing
 /// kernels in the future.
-pub fn infer_json_schema_from_iterator<I>(value_iter: I) -> Result<Schema, ArrowError>
+pub fn infer_json_schema_from_iterator<I, V>(value_iter: I) -> Result<Schema, ArrowError>
 where
-    I: Iterator<Item = Result<Value, ArrowError>>,
+    I: Iterator<Item = Result<V, ArrowError>>,
+    V: AsRef<Value>,
```

Comment:
```suggestion
    V: Borrow<Value>,
```

I think this should remove the need for the boxing